plugins/outputs/file/file.go

```diff
@@ -13,24 +13,64 @@ import (
 	"github.com/influxdata/telegraf/internal/rotate"
 	"github.com/influxdata/telegraf/plugins/outputs"
 	"github.com/influxdata/telegraf/plugins/serializers"
+	"github.com/klauspost/compress/zstd"
 )
 
 //go:embed sample.conf
 var sampleConfig string
 
+var ValidCompressionAlgorithmLevels = map[string][]int{
+	"zstd": {1, 3, 7, 11},
+}
+
+const (
+	defaultCompressionAlgorithm = "zstd"
+	defaultCompressionLevel     = 3
+)
+
 type File struct {
 	Files               []string        `toml:"files"`
 	RotationInterval    config.Duration `toml:"rotation_interval"`
 	RotationMaxSize     config.Size     `toml:"rotation_max_size"`
 	RotationMaxArchives int             `toml:"rotation_max_archives"`
 	UseBatchFormat      bool            `toml:"use_batch_format"`
+	Compression         Compression     `toml:"compression"`
+	Encoder             interface{}
 	Log                 telegraf.Logger `toml:"-"`
 
 	writer     io.Writer
 	closers    []io.Closer
 	serializer serializers.Serializer
 }
 
+type Compression struct {
+	Enabled   bool   `toml:"enabled"`
+	Algorithm string `toml:"algorithm"`
+	Level     int    `toml:"level"`
+}
+
+func ValidateCompressionAlgorithm(algorithm string) error {
```

Comment:
```suggestion
func validateCompressionAlgorithm(algorithm string) error {
```
Same here, I don't think these functions need to be exported?